src/main/scala/io/circe/yaml/Parser.scala

```diff
@@ -29,37 +24,51 @@ final case class Parser(
     json <- yamlToJson(parsed)
   } yield json
 
-  def parse(yaml: String): Either[ParsingFailure, Json] = parse(new StringReader(yaml))
+  def parse(yaml: String): Either[ParsingFailure, Json] =
+    parse(new StringReader(yaml))
 
   def parseDocuments(yaml: Reader): Stream[Either[ParsingFailure, Json]] = parseStream(yaml).map(yamlToJson)
   def parseDocuments(yaml: String): Stream[Either[ParsingFailure, Json]] = parseDocuments(new StringReader(yaml))
 
-  private[this] def parseSingle(reader: Reader) =
-    Either.catchNonFatal(new Yaml(loaderOptions).compose(reader)).leftMap(err => ParsingFailure(err.getMessage, err))
+  private[this] def asScala[T](ot: Optional[T]): Option[T] =
+    if (ot.isPresent) Some(ot.get()) else None
+
+  private[this] def createComposer(reader: Reader) =
+    new Composer(settings, new ParserImpl(settings, new StreamReader(settings, reader)))
```

Comment:
```scala
  new Compose(settings).composeReader(reader)
```
There's a utl method that defines all this + the `.getSingleNode` call in the form of [Compose.composeReader](
https://javadoc.io/static/org.snakeyaml/snakeyaml-engine/2.5/org/snakeyaml/engine/v2/api/lowlevel/Compose.html#composeReader-java.io.Reader-)